client/mm/mm_basic.go

```diff
@@ -411,6 +435,52 @@ func (m *basicMarketMaker) ordersToPlace(basisPrice, feeAdj uint64) (buyOrders,
 
 	buyOrders = orders(m.cfg().BuyPlacements, false)
 	sellOrders = orders(m.cfg().SellPlacements, true)
+	return buyOrders, sellOrders, oracleOutsideSafeRange, nil
+}
+
+// updateBotProblems updates the bot problems based on the errors encountered
+// during the bot loop.
+func (m *basicMarketMaker) updateBotLoopProblems(buyErr, sellErr, determinePlacementsErr error, outsideOracleSafeRange bool, dexDefs map[uint32]uint64) {
+	m.unifiedExchangeAdaptor.updateBotProblems(func(problems *BotProblems) {
+		clearBotProblemErrors(problems)
+		problems.MidGapOutsideOracleSafeRange = outsideOracleSafeRange
+
+		if !updateBotProblemsBasedOnError(problems, buyErr) {
+			problems.PlaceBuyOrdersErr = buyErr
+		}
+
+		if !updateBotProblemsBasedOnError(problems, sellErr) {
+			problems.PlaceSellOrdersErr = sellErr
+		}
+
+		if !updateBotProblemsBasedOnError(problems, determinePlacementsErr) {
+			problems.DeterminePlacementsErr = determinePlacementsErr
+		}
+
+		problems.DEXBalanceDeficiencies = dexDefs
+	})
+}
+
+// clearAllOrders returns slices of order placements with zero rate and lots.
+// When these are passed to multiTrade, all orders will be cancelled.
+func clearAllOrders(numBuyOrders, numSellOrders int) (buyOrders, sellOrders []*multiTradePlacement) {
```

Comment:
Woudn't it be easier to create a method that just cancels all of the orders? What's the advantage of plumbing the through `mutliTrade`?